test/src/main/java/com/redhat/lightblue/mongo/test/AbstractMongoCRUDTestController.java

```diff
@@ -0,0 +1,100 @@
+/*
+ Copyright 2015 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of lightblue.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue.mongo.test;
+
+import static com.redhat.lightblue.util.JsonUtils.json;
+
+import java.io.IOException;
+import java.net.UnknownHostException;
+
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.redhat.lightblue.mongo.test.MongoServerExternalResource.InMemoryMongoServer;
+import com.redhat.lightblue.test.AbstractCRUDTestController;
+
+/**
+ * <p>Extension of {@link AbstractCRUDTestController} that adds built in mongo support.</p>
+ * <p><b>NOTE:</b> At this time only restarting mongo before/after the suite is supported.
+ * That said, if you need to clean the mongo database between tests then see {@link #cleanupMongoCollections(String...)}. </p>
+ *
+ * @author dcrissman
+ */
+@InMemoryMongoServer
+public abstract class AbstractMongoCRUDTestController extends AbstractCRUDTestController {
+
+    @ClassRule
+    public static MongoServerExternalResource mongoServer = new MongoServerExternalResource();
+
+    @BeforeClass
+    public static void prepareMongoDatasources() {
+        if (System.getProperty("mongo.host") == null) {
+            System.setProperty("mongo.host", "localhost");
+        }
+        if (System.getProperty("mongo.port") == null) {
+            System.setProperty("mongo.port", String.valueOf(mongoServer.getPort()));
+        }
+        if (System.getProperty("mongo.database") == null) {
+            System.setProperty("mongo.database", "lightblue");
+        }
+    }
+
+    public AbstractMongoCRUDTestController() throws Exception {
+        super(true);
+    }
+
+    @Override
+    protected JsonNode getDatasourcesJson() {
+        try {
+            if (getDatasourcesResourceName() == null) {
+                return json(loadResource("/mongo-datasources.json", true));
```

Comment:
Honestly, I am not sure. Both ways seem to work. I followed @paterczm (name drop alert) example  over in https://github.com/lightblue-platform/lightblue-core/blob/master/test/src/main/java/com/redhat/lightblue/test/AbstractCRUDTestController.java#L108

I am happy to go either direction.